lib/dry/schema/macros/key.rb

```diff
@@ -6,6 +6,16 @@ module Dry
   module Schema
     module Macros
       class Key < DSL
+        TYPE_TO_PREDICATE = ::Hash.new { |hash, type|
+          primitive = type.maybe? ? type.right.primitive : type.primitive
+
+          if hash.key?(primitive)
+            hash[primitive]
+          else
+            :"#{primitive.name.downcase}?"
```

Comment:
I wonder if we want this hash to act somewhat like a cache, and have it save entries inside itself whenever it finds a match?